common/TruffleConfig.js

```diff
@@ -75,9 +74,9 @@ function addPublicNetwork(networks, name, networkId) {
   // Mnemonic network.
   networks[name + "_mnemonic"] = {
     ...options,
-    provider: function() {
+    provider: function(provider = nodeUrl) {
       if (!singletonProvider) {
-        singletonProvider = new HDWalletProvider(mnemonic, nodeUrl, keyOffset, numKeys);
+        singletonProvider = new HDWalletProvider(mnemonic, provider, keyOffset, numKeys);
```

Comment:
Something to think about -- if you call this function twice, it will ignore your provider input on the second call and just give you the singleton. Probably not something to worry too much about on this PR, but we may want to rework this a little so you can call it without invoking the singleton workaround built for truffle.

Reply:
yup, for sure. it could be restrictive. for now I don't see us needing multiple providers so probs fine.